glotaran/analysis/test/models.py

```diff
@@ -56,13 +56,22 @@ def index_dependent(self, dataset_model):
 
 class SimpleTestModel(Model):
     @classmethod
-    def from_dict(cls, model_dict):
-        return super().from_dict(
-            model_dict,
-            megacomplex_types={
+    def from_dict(
+        cls,
+        model_dict,
+        *,
+        megacomplex_types: dict[str, type[Megacomplex]] | None = None,
+        default_megacomplex_type: str | None = None,
+    ):
+        if megacomplex_types is None:
```

Comment:
If `megacomplex_type` is not `None` the default types should be updated like

```python
defaults = {
    "model_complex": SimpleTestMegacomplex,
    "global_complex": SimpleTestMegacomplexGlobal,
}
if megacomplex_types is not None:
    defaults.update(megacomplex_types)
```

Reply:
Implemented the suggestion in [b6f6833](https://github.com/joernweissenborn/pyglotaran/pull/4/commits/b6f6833d399ce324aa4ffab381bb5f51966be86f)